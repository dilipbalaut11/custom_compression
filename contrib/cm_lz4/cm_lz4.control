# cm_lz4 extension
comment = 'cm_lz4 compression method'
default_version = '1.0'
module_pathname = '$libdir/cm_lz4'
relocatable = true
