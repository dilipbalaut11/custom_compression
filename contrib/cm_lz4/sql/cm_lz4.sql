CREATE EXTENSION cm_lz4;

-- zlib compression
CREATE TABLE lz4test(f1 TEXT COMPRESSION lz4);
INSERT INTO lz4test VALUES(repeat('1234567890',1004));
INSERT INTO lz4test VALUES(repeat('1234567890 one two three',1004));
SELECT length(f1) FROM lz4test;

-- alter compression method with rewrite
ALTER TABLE lz4test ALTER COLUMN f1 SET COMPRESSION zlib;
\d+ lz4test
ALTER TABLE lz4test ALTER COLUMN f1 SET COMPRESSION lz4;
\d+ lz4test

-- preserve old compression method
ALTER TABLE lz4test ALTER COLUMN f1 SET COMPRESSION zlib PRESERVE (lz4);
INSERT INTO lz4test VALUES (repeat('1234567890',1004));
\d+ lz4test
SELECT length(f1) FROM lz4test;

DROP TABLE lz4test;
