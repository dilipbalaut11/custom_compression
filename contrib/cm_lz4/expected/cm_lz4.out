CREATE EXTENSION cm_lz4;
-- zlib compression
CREATE TABLE lz4test(f1 TEXT COMPRESSION lz4);
INSERT INTO lz4test VALUES(repeat('1234567890',1004));
INSERT INTO lz4test VALUES(repeat('1234567890 one two three',1004));
SELECT length(f1) FROM lz4test;
 length 
--------
  10040
  24096
(2 rows)

-- alter compression method with rewrite
ALTER TABLE lz4test ALTER COLUMN f1 SET COMPRESSION zlib;
\d+ lz4test
                                        Table "public.lz4test"
 Column | Type | Collation | Nullable | Default | Storage  | Compression | Stats target | Description 
--------+------+-----------+----------+---------+----------+-------------+--------------+-------------
 f1     | text |           |          |         | extended | zlib        |              | 

ALTER TABLE lz4test ALTER COLUMN f1 SET COMPRESSION lz4;
\d+ lz4test
                                        Table "public.lz4test"
 Column | Type | Collation | Nullable | Default | Storage  | Compression | Stats target | Description 
--------+------+-----------+----------+---------+----------+-------------+--------------+-------------
 f1     | text |           |          |         | extended | lz4         |              | 

-- preserve old compression method
ALTER TABLE lz4test ALTER COLUMN f1 SET COMPRESSION zlib PRESERVE (lz4);
INSERT INTO lz4test VALUES (repeat('1234567890',1004));
\d+ lz4test
                                        Table "public.lz4test"
 Column | Type | Collation | Nullable | Default | Storage  | Compression | Stats target | Description 
--------+------+-----------+----------+---------+----------+-------------+--------------+-------------
 f1     | text |           |          |         | extended | zlib        |              | 

SELECT length(f1) FROM lz4test;
 length 
--------
  10040
  24096
  10040
(3 rows)

DROP TABLE lz4test;
